pkg/statistics/handle/ddl/drop_partition.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ddl
+
+import (
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tidb/pkg/sessionctx"
+	"github.com/pingcap/tidb/pkg/sessionctx/variable"
+	"github.com/pingcap/tidb/pkg/statistics/handle/lockstats"
+	"github.com/pingcap/tidb/pkg/statistics/handle/storage"
+	"github.com/pingcap/tidb/pkg/statistics/handle/util"
+)
+
+func (h *ddlHandlerImpl) onDropPartitions(t *util.DDLEvent) error {
+	globalTableInfo, droppedPartitionInfo := t.GetDropPartitionInfo()
+	// Note: Put all the operations in a transaction.
+	return util.CallWithSCtx(h.statsHandler.SPool(), func(sctx sessionctx.Context) error {
+		count := int64(0)
+		for _, def := range droppedPartitionInfo.Definitions {
+			// Get the count and modify count of the partition.
+			tableCount, _, _, err := storage.StatsMetaCountAndModifyCount(sctx, def.ID)
```

Comment:
Do we need to subtract the modify count from partition stats when we update the modify count as we did in the exchange partition?

Reply:
Considering the exchange partition logic, it seems we should it as `update + delete`. So maybe we should keep it?

Reply:
In my understanding, in the exchange partition logic, the modify count update consists of two parts when we remove a partition from the partitioned table:
1. We consider that we deleted this table, so we should add the row count of the partition to the global modify count.
2. Some of the original global modify count is contributed by this partition. Now that this partition is removed, we need to subtract this modify count.

Reply:
> 2\. Some of the original global modify count is contributed by this partition. Now that this partition is removed, we need to subtract this modify count.

But for dropped partitions, it happened before it got dropped. So I think it should count as changes to the global table. For instance, this situation more like: 
1. The partition has 1000 rows.
2. I updated 500 rows.
3. I deleted 999 rows.

"Should we consider the modify count as 500 plus 999, or just 999?"

Regarding the exchange partition logic, we include the modify count of the exchange table in the global stats. 
https://github.com/pingcap/tidb/blob/cf23666766d03ec9c2551465fbb92e0f9bba25a2/pkg/statistics/handle/ddl/exchange_partition.go#L52
Here we not only subtract the old partition's modify count, but we also add the new table's modify count.

Reply:
I didn't fully understand your meaning.
For the exchange partition formula, I think the removing partition contributes to the `partCount - partModifyCount` part, and the adding partition contributes to the `tableCount + tableModifyCount` part. So I think we missed the ` - partModifyCount` part here.

Reply:
After we discussed offline: We shouldn't drop those updates, because it happens before the drop operation.

For example: 
1. The partition has 1000 rows.
2. I deleted 1000 rows.
3. I dropped the partition.

We shouldn't update the global table's modify count to 0.